go/ql/lib/semmle/go/frameworks/stdlib/Regexp.qll

```diff
@@ -6,13 +6,41 @@ import go
 
 /** Provides models of commonly used functions in the `regexp` package. */
 module Regexp {
-  private class Pattern extends RegexpPattern::Range, DataFlow::ArgumentNode {
-    string fnName;
+  /**
+   * Holds if `kind` is an external sink kind that is relevant for regex flow.
+   * `strArg` is the index of the argument to methods with this sink kind that
+   * contain the string to be matched against, where -1 is the qualifier; or -2
+   * if no such argument exists and the function compiles the regex; or -3 if
+   * no such argument exists and the function does not compile the regex.
+   *
+   * So `regex-use[0]` indicates that argument 0 contains the string to matched
+   * against, `regex-use[c]` indicates that there is no string to be matched
+   * against and the return value of the function is a compiled regex, and
+   * `regex-use` means that there is no string to be matched against and the
+   * function does not compile the regex.
+   */
+  private predicate regexSinkKindInfo(string kind, int strArg) {
+    strArg = -3 and
+    kind = "regex-use"
+    or
+    sinkModel(_, _, _, _, _, _, _, kind, _, _) and
+    exists(string strArgStr |
+      strArgStr.toInt() = strArg
+      or
+      strArg = -2 and
+      strArgStr = "c"
+    |
+      kind = "regex-use[" + strArgStr + "]"
+    )
```

Comment:
Does this need to add handling of `-1` corresponding to `receiver` like done in the http concepts library?